pipelines/build/common/openjdk_build_pipeline.groovy

```diff
@@ -390,11 +392,91 @@ class Build {
                 .toList()
     }
 
-    MetaData formMetadata(VersionInfo version) {
-        return new MetaData(buildConfig.TARGET_OS, buildConfig.SCM_REF, version, buildConfig.JAVA_TO_BUILD, buildConfig.VARIANT, buildConfig.ARCHITECTURE)
+    MetaData formMetadata(VersionInfo version, Boolean initialWrite) {
+
+        if (initialWrite) {
```

Comment:
from what I can see of this section it could probably be simplified to something like:

```
scmRef=buildConfig.SCM_REF
        
        if (initialWrite) {

            String scmRefPath = "workspace/target/scmref.txt"

            context.println "FIRST METADATA WRITE OUT! Checking if we have a scm reference in the build config..."

            // Use the buildConfig scmref if it exists
            if (buildConfig.SCM_REF != "") {
                context.println "SCM_REF has been set (${buildConfig.SCM_REF})! Using it to build the inital metadata over ${scmRefPath}..."
                scmRef=buildConfig.SCM_REF
            } else {
                // If we don't have a scmref set in config, check if we have a scmref from the build
                context.println "SCM_REF is not set. Attempting to read ${scmRefPath}..."
                try {
                    scmRef = context.readFile(scmRefPath).trim()
                } catch (NoSuchFileException e) {
                    // In rare cases, we will fail to create the scmref.txt file
                    context.println "WARNING: $scmRefPath was not found."
                }
            }
        }
        
        return new MetaData(
            buildConfig.TARGET_OS,
            scmRef,
            version,
            buildConfig.JAVA_TO_BUILD,
            buildConfig.VARIANT,
            buildConfig.ARCHITECTURE
        )
```

Reply:
I'm not sure I agree with setting `scmRef` to never be an empty string. In later writeouts, when `initialWrite` is set to false, `scmRef` will never be an empty string. So if the user has specified a tag to build, it will never be put into the metadata

Reply:
from what I can tell this would mirror the functionality you already have, I believe in all paths in your code scmref is set to either `buildConfig.SCM_REF` or the contents of the scmRefPath file

Reply:
Looking at it again, I've found a way to make it work (I think)! @johnoliver Can you check the latest commit?

Reply:
yup looks good, although you can probably go further as both the if and else return the same thing on 420 and 430

Reply:
Agreed